crates/paimon/src/spec/schema.rs

```diff
@@ -51,3 +51,133 @@ pub struct DataField {
     typ: DataType,
     description: Option<String>,
 }
+
+impl DataField {
+    pub fn new(id: i32, name: String, typ: DataType) -> Self {
+        Self {
+            id,
+            name,
+            typ,
+            description: None,
+        }
+    }
+
+    pub fn id(&self) -> i32 {
+        self.id
+    }
+
+    pub fn name(&self) -> &str {
+        &self.name
+    }
+
+    pub fn data_type(&self) -> &DataType {
+        &self.typ
+    }
+
+    pub fn description(&self) -> Option<&String> {
+        self.description.as_ref()
+    }
+
+    pub fn with_id(mut self, new_id: i32) -> Self {
+        self.id = new_id;
+        self
+    }
+
+    pub fn with_name(mut self, new_name: String) -> Self {
+        self.name = new_name;
+        self
+    }
+
+    pub fn with_description(mut self, new_description: String) -> Self {
```

Comment:
I personal feel like it's better to have an `Option<String>` here so users can remove `description` if needed. 

It's also possible to have `impl Into<Option<String>>`, but I feel it's a bit over-designed for now.

Reply:
I also think that impl Into<Option<String>> might be more appropriate if needed in the future, but I agree with using Option<String> for now.

Reply:
done